go/autoscaler/k8s/controller.go

```diff
@@ -48,21 +48,10 @@ type Controller struct {
 }
 
 // NewController construct a new Controller struct
-func NewController(config *rest.Config) (*Controller, error) {
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, err
-	}
-	client, err := rest.RESTClientFor(config)
-	if err != nil {
-		return nil, err
-	}
-	// TODO: init autoscaler with correct arguments.
-	cluster := autoscaler.NewK8sCluster(clientset)
-	as := autoscaler.NewAutoscaler(cluster)
+func NewController(c *rest.RESTClient, cs *kubernetes.Clientset, as *autoscaler.Autoscaler) (*Controller, error) {
```

Comment:
"Dependency injection": pass in the dependencies, rather than creating dependency. This can simplify testing.